tests/test_qasm_examples.py

```diff
@@ -1,17 +1,136 @@
 import os
 import re
+import sympy
 import pytest
+import numpy as np
 
 from qdata import parser
 
+
+def split_at_semicol(qasm_lines):
+    """Helper function to split lines at semicolons"""
+    form_lines = []
+    for line_1 in qasm_lines:
+        split_lines = line_1.split(";")
+        if len(split_lines) == 1:
+            form_lines.append(split_lines[0])
+        elif len(split_lines) == 2 and split_lines[1] == "":
+            form_lines.append(split_lines[0] + ";")
+        else:
+            for line_2 in split_lines:
+                delim = ";" * (clean(line_2) not in ["{", "}"])
+                if clean(line_2):
+                    form_lines.append(clean(line_2) + delim)
+    return form_lines
+
+
+def fix_braces(qasm_lines):
+    """Helper function to place braces on separate lines"""
+    form_lines = []
+    for line_1 in qasm_lines:
+        # if a line contains empty braces, format them separately
+        match = re.search(r"{\s*}", line_1)
+        if match:
+            form_lines.append(line_1.replace(match.group(), "{ }"))
+            continue
+
+        # split at "{" and re-add them as separate list-elements
+        res = line_1.split("{")
+        next_qasm_lines = [clean(em) for el in list(zip(["{"]*len(res), res)) for em in el if clean(em)][1:]
+        for line_2 in next_qasm_lines:
+            # split the new list at "}" and re-add them as separate list-elements
+            res = line_2.split("}")
+            lines = [clean(em) for el in list(zip(["}"]*len(res), res)) for em in el if clean(em)][1:]
+            form_lines.extend(lines)
+
+    #asdad
+    tab = False
+    for i in range(1, len(form_lines) - 1):
+        if tab and form_lines[i] != "}":
+            form_lines[i] = "    " + form_lines[i]
+
+        if form_lines[i] == "{":
+            tab = True
+        elif form_lines[i] == "}":
+            tab = False
```

Comment:
This variable could be named a bit better, perhaps `indented_block`. That way, the if statement would be `if indented_block and form_lines...`